### PR TITLE
Create locale list page

### PR DESCRIPTION
**Pull request by MickJasker**

### CODE REVIEW

File: src/store/projects.ts [hunk 1]
```diff
@@ -17,18 +17,33 @@ interface ProjectRawMetadata {
   name: string;
   id: string;
   users: string[];
+  locales?: Array<{
+    name: string;
+    code: string;
+  }>;
 }
 
 interface ProjectMetadata {
```
Comment (ThijsTyZ) [inline src/store/projects.ts:26]: Why do you have 2 almost identical interfaces?

File: src/store/projects.ts [hunk 2]
```diff
@@ -17,18 +17,33 @@ interface ProjectRawMetadata {
...
   users: User[];
+  locales: Array<{
+    name: string;
+    code: string;
+  }>;
+}
+
+interface ProjectLocale {
+  name: string;
+  content?: any[] | Record<string, any>;
```
Comment (ThijsTyZ) [inline src/store/projects.ts:38]: avoid `any`

File: src/store/projects.ts [hunk 3]
```diff
@@ -17,18 +17,33 @@ interface ProjectRawMetadata {
...
 }
 
 interface Project {
   metadata?: ProjectMetadata;
   schemaURL?: string;
-  content?: any[] | Record<any, any>;
+  locales?: {
+    [key: string]: ProjectLocale;
```
Comment (ThijsTyZ) [inline src/store/projects.ts:45]: Change this to `Record<string, ProjectLocale>`

File: src/store/projects.ts [hunk 4]
```diff
@@ -71,6 +86,7 @@ async function getFormattedProjects(ids: string[]): Promise<ProjectMetadata[]> {
       name: rawProject.name,
       id: rawProject.id,
       users: projectsUsers,
+      locales: rawProject.locales ? rawProject.locales : [],
```
Comment (ThijsTyZ) [inline src/store/projects.ts:89]: you could shorten this to:
```
locales: rawProject.locales || [],
```

File: src/store/projects.ts [hunk 5]
```diff
@@ -166,22 +182,22 @@ const fetchJSONSchema = async (url: string) => {
...
 
-  const projectMetadata = await getFormattedProjects([id]);
-
   projectRef.on('value', async (snapshot) => {
     const data: null | {
       schemaURL?: string;
-      content?: any[] | {
-        [key: string]: any;
+      locales?: {
+        [key: string]: ProjectLocale;
```
Comment (ThijsTyZ) [inline src/store/projects.ts:195]: Change to `Record`

File: src/store/projects.ts [hunk 6]
```diff
@@ -271,14 +289,74 @@ async function updateProjectsMetadata(newMetadata: ProjectMetadata): Promise<Pro
...
+  return projectsState.currentProject.locales[code].content;
+}
+
+function downloadData(data: object | any[], name = 'content') {
+  const dataString = `data:text/json;charset=utf-8,${encodeURIComponent(JSON.stringify(data))}`;
+  const anchorNode = document.createElement('a');
+  anchorNode.setAttribute('style', 'display: hidden;');
+  anchorNode.setAttribute('href', dataString);
+  anchorNode.setAttribute('download', `${name}.json`);
+  document.body.appendChild(anchorNode);
```
Comment (ThijsTyZ) [inline src/store/projects.ts:342]: I don't think you need to add it to the DOM in order to get it functional

File: src/views/project/Content.vue [hunk 7]
```diff
@@ -55,13 +91,46 @@ export default defineComponent({
...
+    locale: string;
+  }) {
     const jsonEditor = ref<HTMLDivElement | null>(null);
     const contentData = ref<object | null>(null);
     let editor: any = null;
     let mdEditors: EasyMDE[] = [];
 
+    const projectData = computed<object | any[] | null>((): any => {
+      const { locale } = props;
+      if (!projectsState.currentProject) return null;
```
Comment (ThijsTyZ) [inline src/views/project/Content.vue:112]: Maybe you should enable optional chaining

File: src/views/project/Content.vue [hunk 8]
```diff
@@ -85,6 +163,52 @@ export default defineComponent({
...
 
+    function resetMarkdownEditors() {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          mdEditors.forEach((mdEditor) => {
+            mdEditor.toTextArea();
+            // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+            // @ts-ignore
+            // eslint-disable-next-line no-param-reassign
+            mdEditor = null;
```
Comment (ThijsTyZ) [inline src/views/project/Content.vue:174]: why is this?

File: src/views/project/Content.vue [hunk 9]
```diff
@@ -144,32 +250,71 @@ export default defineComponent({
...
       });
     }
 
+    const referenceLocale = ref<string | null>(null);
+
+    watch(referenceLocale, (value) => {
+      if (!jsonEditor.value) return;
+      const currentRefElements = [...jsonEditor.value.querySelectorAll<HTMLDivElement>('.ref-element')];
+      currentRefElements.forEach((element) => {
+        // eslint-disable-next-line no-param-reassign
```
Comment (ThijsTyZ) [inline src/views/project/Content.vue:275]: are you sure you need to add this line? you are not reassigning a param

File: src/views/project/Content.vue [hunk 10]
```diff
@@ -144,32 +250,71 @@ export default defineComponent({
...
+        const inputEl = element.querySelector<
+          HTMLInputElement |
+          HTMLTextAreaElement |
+          HTMLSelectElement>('[name]');
+        let path = inputEl?.getAttribute('name');
+
+        if (!path) return;
+
+        const filterWords = ['root'];
+        const rgx = new RegExp(filterWords.join(''), 'gi');
```
Comment (ThijsTyZ) [inline src/views/project/Content.vue:295]: can you add a comment why you are doing this? code is not clear

File: src/views/project/Content.vue (hunk 9 above)
Comment (MickJasker) [reply src/views/project/Content.vue:275]: EsLint thinks it is:
`271:9  error  Assignment to property of function parameter 'element'  no-param-reassign`
